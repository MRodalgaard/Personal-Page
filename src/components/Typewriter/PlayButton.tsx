import React from 'react';
import styled from 'styled-components';

const StyledPlayButton = styled.i`
  float: right;
  cursor: pointer;
  transition: color 0.5s ease;

  &:hover {
    color: ${({ theme }) => theme.secondary};
  }

  animation: cursor 0.5s linear infinite alternate;

  @keyframes cursor {
    0% {
      opacity: 0.1;
    }
    100% {
      opacity: 1;
    }
  }

  & {
    box-sizing: border-box;
    position: relative;
    display: block;
    transform: scale(var(--ggs, 1));
    width: 6px;
    height: 10px;
  }
  &::before {
    content: '';
    display: block;
    box-sizing: border-box;
    position: absolute;
    width: 0;
    height: 10px;
    border-top: 5px solid transparent;
    border-bottom: 5px solid transparent;
    border-left: 6px solid;
  }
`;

interface IProps {
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
}

const PlayButton = ({ onClick }: IProps) => {
  return <StyledPlayButton onClick={onClick}></StyledPlayButton>;
};

export default PlayButton;
