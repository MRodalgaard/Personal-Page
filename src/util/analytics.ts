import 'firebase/analytics';
import firebase from './firebase';

let analytics: firebase.analytics.Analytics | undefined;

export enum LogEvent {
  ColorClick = 'color_click',
  QuoteClick = 'quote_click',
}

class Analytics {
  constructor() {
    firebase.analytics.isSupported().then((isSupported) => {
      if (isSupported) {
        analytics = firebase.analytics();
      }
    });
  }

  public logEvent = (event: LogEvent, parameters?: any) => {
    analytics?.logEvent(event, parameters);
  };
}

export default new Analytics();
