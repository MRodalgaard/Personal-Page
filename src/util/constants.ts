// Key for persisting app state in local storage
export const LOCALSTORAGE_KEY = 'AppState';

export const DEFAULT_AGE = 30;
export const TYPEWRITER_DELAY = 30;

export enum AppBackground {
  greyscale,
  colored,
}
