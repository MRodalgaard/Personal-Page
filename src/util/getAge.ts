const getAge = (birthday: string): number | undefined => {
  const today = new Date();
  const birthDate = new Date(birthday);

  if (isNaN(birthDate.getTime()) || today < birthDate) {
    return undefined;
  }

  let age = today.getFullYear() - birthDate.getFullYear();
  const m = today.getMonth() - birthDate.getMonth();
  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
    age--;
  }
  return age;
};

export default getAge;
