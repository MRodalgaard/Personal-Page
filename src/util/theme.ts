export enum AppColors {
  primary = '#000000',
  secondary = '#c9282d',
  grey = '#333333',
}
