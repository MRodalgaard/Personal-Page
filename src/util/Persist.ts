import { IAppState } from '../components/App/AppContext';
import { LOCALSTORAGE_KEY } from './constants';

const Persist = {
  persistState: (state: IAppState) => {
    localStorage.setItem(LOCALSTORAGE_KEY, JSON.stringify(state));
  },

  getPersistedState: (): IAppState | undefined => {
    const jsonState = localStorage.getItem(LOCALSTORAGE_KEY);
    return jsonState ? JSON.parse(jsonState) : undefined;
  },
};

export default Persist;
