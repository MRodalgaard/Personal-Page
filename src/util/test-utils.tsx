import { render, RenderOptions } from '@testing-library/react';
import AppProvider from 'components/App/AppProvider';
import React, { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import theme from './theme';

type Props = {
  children: ReactNode;
};

const AllTheProviders = ({ children }: Props) => {
  return (
    <AppProvider>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </AppProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, {
    wrapper: AllTheProviders as React.FunctionComponent,
    ...options,
  });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
