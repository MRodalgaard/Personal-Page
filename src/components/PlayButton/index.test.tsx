import React from 'react';
import { render } from 'util/test-utils';
import PlayButton from '.';

describe('PlayButton', () => {
  it('renders', () => {
    const { container } = render(<PlayButton></PlayButton>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
