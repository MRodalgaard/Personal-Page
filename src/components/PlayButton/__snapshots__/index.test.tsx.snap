// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayButton renders 1`] = `
.c0 {
  float: right;
  cursor: pointer;
  -webkit-transition: color 0.5s ease;
  transition: color 0.5s ease;
  -webkit-animation: cursor 0.5s linear infinite alternate;
  animation: cursor 0.5s linear infinite alternate;
}

.c0:hover {
  color: #c9282d;
}

.c0 {
  box-sizing: border-box;
  position: relative;
  display: block;
  -webkit-transform: scale(var(--ggs,1));
  -ms-transform: scale(var(--ggs,1));
  transform: scale(var(--ggs,1));
  width: 6px;
  height: 10px;
}

.c0::before {
  content: '';
  display: block;
  box-sizing: border-box;
  position: absolute;
  width: 0;
  height: 10px;
  border-top: 5px solid transparent;
  border-bottom: 5px solid transparent;
  border-left: 6px solid;
}

<i
  class="c0"
/>
`;
